### PR TITLE
Passcode lock UI

### PR DESCRIPTION
**Pull request by ChaoticWeg**
- User should be able to swipe up to reveal an overlaying panel
 - User can enter passcode
 - When 4 digits are entered, panel is dismissed and main settings screen is presented
 - UI exists for displaying an error if passcode is incorrect
 - For now, panel can be extended all the way to the top

### CODE REVIEW

File: src/screens/PasscodeGateScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+*
+*   eventually we can refactor this into a Modal, which would probably be the best option, but this is all
+*   i have time to do right now. sorry for this terrible hack.
+*
+*   should ABSOLUTELY UNDER NO CIRCUMSTANCES hard-code the title like this for something of this nature,
+*   but since we're only using this in one specific instance i've done it to save time.
+*
+*   jesus christ this hack is god awful
+*
+*   - shawn, 25 apr 2019 (23:30, night before demo)
```
Comment (richardkriesman) [inline src/screens/PasscodeGateScreen.tsx:30]: > 23:30, night before demo

Beautiful lol

File: src/typings/index.d.ts [hunk 2]
```diff
@@ -32,5 +32,8 @@ declare module "@expo/samples" {
 // getting an error on my end. hoping to clear it up. sL
 declare module "@expo/vector-icons";
 
+// Getting an error, pls work
+declare module "react-native-vertical-swipe";
```
Comment (richardkriesman) [inline src/typings/index.d.ts:36]: Obviously, I assume this will be fixed after the demo, right? :p

**Review by richardkriesman — CHANGES_REQUESTED** [review]
Okay, so apart from the terrible hacks (which we can fix after the demo) the one issue I had is in clickthrough testing: if the user does not have a passcode set, the gate still appears and there's no way to get into the settings screen.

File: src/typings/index.d.ts (hunk 2 above)
Comment (ChaoticWeg) [reply src/typings/index.d.ts:36]: That was cody :P

Comment (richardkriesman) [reply src/typings/index.d.ts:36]: I'm callin u out @Kyrk

**Comment by ChaoticWeg** [comment]
> 
> 
> Okay, so apart from the terrible hacks (which we can fix after the demo) the one issue I had is in clickthrough testing: if the user does not have a passcode set, the gate still appears and there's no way to get into the settings screen.

Will fix.

**Review by richardkriesman — APPROVED** [review]

**Pull request merged**